### PR TITLE
[internals] Create `@vercel-internals/types`

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Moves the type file out of the cli package and into its own standalone package. utilizes `@vercel/style-guide` too for typescript config, eslint, and prettier.

### CODE REVIEW

File: internals/types/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@vercel-internals/types",
+  "private": true,
+  "types": "index.d.ts",
+  "dependencies": {
+    "@vercel/build-utils": "workspace:6.3.2",
+    "@vercel/routing-utils": "workspace:2.1.10"
```
Comment (styfle) [inline internals/types/package.json:7]: Didn't this syntax break last time?

File: packages/cli/package.json [hunk 2]
```diff
@@ -93,6 +93,7 @@
     "@types/which": "1.3.2",
     "@types/write-json-file": "2.2.1",
     "@types/yauzl-promise": "2.1.0",
+    "@vercel-internals/types": "workspace:*",
```
Comment (styfle) [inline packages/cli/package.json]: Didn't this syntax break last time?

Comment (Ethan-Arrowood) [reply packages/cli/package.json]: I think so. But I forget what it broke so I wanted to see if we could fix that this time around.

File: internals/types/package.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@vercel-internals/types",
+  "private": true,
```
Comment (styfle) [inline internals/types/package.json:2]: Nit pick: put `private` as the first property in the file

File: packages/cli/package.json (hunk 2 above)
Comment (EndangeredMassa) [reply packages/cli/package.json]: This broke the published package, if I recall correctly

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**